contracts/OriginationController.sol

```diff
@@ -174,6 +174,12 @@ contract OriginationController is
 
         (bytes32 sighash, address externalSigner) = recoverTokenSignature(loanTerms, sig, nonce, neededSide);
 
+        if (externalSigner == borrower) {
```

Comment:
Do we also need this logic in `initializeLoanWithItems`? Seems like it should maybe go inside `_validateCounterparties` to cover the most bases

Reply:
Also: revert with a custom error here, as opposed to using `require`? to be consistent with the rest of the file